File: Composicion y Agregacion/Ejercicio9/Java/Producto.java
import java.util.ArrayList;
class Prodúcto {
    public String nombre;
    public double precio;
    public Prodúcto(String nombre, double precio) {
        this.nombre = nombre;
        this.precio = precio;
    }
    public void mostrar() {
        System.out.println("Prodúcto: " + nombre + ", Precio: BOB" + precio);
    }
}
class CarritoCompras {
    public ArrayList<Prodúcto> productos;
    public CarritoCompras() {
        productos = new ArrayList<>();
    }
    public void agrega(Prodúcto p) {
        if (productos.size() < 10) {
            productos.add(p);
        } else {
            System.out.println("No se pueden agregar más productos, carro lleno");
        }
    }
    public void mostrar() {
        System.out.println("Carrito de compras:");
        for (Prodúcto p : productos) {
            p.mostrar();
        }
    }
}
public class Producto {
    public static void main(String[] args) {
        CarritoCompras c = new CarritoCompras();
        c.agrega(new Prodúcto("Leche", 1.5));
        c.agrega(new Prodúcto("Pan", 0.8));
        c.agrega(new Prodúcto("Queso", 2.5));
        c.mostrar();
    }
}

File: Composicion y Agregacion/Ejercicio9/Python/Producto.py
class Producto:
    def __init__(self, nombre, precio):
        self.nombre = nombre
        self.precio = precio
    def mostrar(self):
        print("Producto:", self.nombre)
        print("Precio:", self.precio)
class CarritoCompras:
    def __init__(self):
        self.productos = []
    def agregar(self, producto):
        if len(self.productos) < 10:
            self.productos.append(producto)
        else:
            print("No se pueden agregar más productos")
    def mostrar(self):
        print("Carrito de Compras:")
        for p in self.productos:
            p.mostrar()
c = CarritoCompras()
for i in range(12):
    c.agregar(Producto("Producto" + str(i+1), i * 10))
c.mostrar()
